L1Trigger/L1TGlobal/plugins/L1TExtCondProducer.cc

```diff
@@ -137,7 +137,7 @@ void L1TExtCondProducer::produce(Event& iEvent, const EventSetup& iSetup) {
   }
 
   bool TriggerRulePrefireVetoBit(false);
-  if (makeTriggerRulePrefireVetoBit_) {
+  if (iEvent.isRealData() && makeTriggerRulePrefireVetoBit_) {
     // code taken from Nick Smith's EventFilter/L1TRawToDigi/plugins/TriggerRulePrefireVetoFilter.cc
 
```

Comment:
@elfontan , thanks for following up on these fixes. Imho, the following change should also be included in `L1TExtCondProducer.cc` to avoid the seg-fault discussed in https://github.com/cms-sw/cmssw/issues/36786#issuecomment-1023479068 , throwing an exception instead:
```diff
    // should be 16 according to TCDSRecord.h, we only care about the last 4
    if (eventHistory.size() < 4) {
-      edm::LogError("L1TExtCondProducer") << "Unexpectedly small L1A history from TCDSRecord";
+      throw cms::Exception("L1TExtCondProducer") << "Unexpectedly small L1A history from TCDSRecord: (size = " << eventHistory.size() << " < 4)";
    }
```
The reason is that `LogError` does not stop the routine, but after that the code assumes that the vector has at least size==4 , and this is what caused the seg-fault.